### ISSUE TITLE
Silly session error

### ISSUE THREAD

**Issue by RWOverdijk** [balderdashy/sails#2385]
Source: https://github.com/balderdashy/sails/blob/master/lib/hooks/session/index.js#L322

```javascript
sails.log.silly('socket.handshake.sessionID is now :: ', socket.handshake.sessionID);
```

This causes my server to restart, here:

```
/home/user/apps/app_name/node_modules/connect-redis/node_modules/redis/index.js:525
            throw err;
                  ^
TypeError: Cannot read property 'sessionID' of undefined
    at /home/user/apps/app_name/node_modules/sails/lib/hooks/session/index.js:322:84
```

So, I'm guessing that at this point, some other code unset the handshake. I could PR with a simple check before the log, but I doubt that'd be enough. We should probably handle that. Calling cb with error would cause an error so, not entirely sure what to do here.

**Update**: 
`Session.generate()` is async, so in between ticks the connection could have dropped and unset the handshake.

**Comment by Dallas62** [comment]
Is it fix now?

**Comment by RWOverdijk** [comment]
I don't know... 0.11 has BC breaks so upgrading takes a small while.

**Comment by sobtiankit** [comment]
Were you able to figure this out? I have been seeing this on my servers as well.

**Comment by RWOverdijk** [comment]
I haven't, no. I proposed a fix but didn't get a reply.

**Comment by RWOverdijk** [comment]
Edit, the code in question is no longer in the source. So I'll update sooner or later to check if it works now.

**Comment by sobtiankit** [comment]
Great. Thanks, will do the same.

**Comment by mikermcneil** [comment]
@RWOverdijk thanks for hunting this down!

@sobtiankit @Dallas62 guys afaik this is resolved in 0.11- would you mind taking a look?  If you have some tips on creating failing tests which reproduce this, I can add them-- I spent a long time working on session/socket tests in the new sockets hook for 0.11 though, so I'm pretty confident this is solid now.

**Comment by sobtiankit** [comment]
I haven't been able to reproduce this in 0.11, at all. I think this can be closed.

**Comment by mikermcneil** [comment]
Thanks for posting, @RWOverdijk.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
